src/System.Private.Uri/tests/FunctionalTests/UriIsWellFormedUriStringTest.cs

```diff
@@ -290,5 +291,182 @@ public void UriIsWellFormed_IPv6HostIriOn_True()
+            
+
+            // Test Path
+            new object[] { "http://www.contoso.com/path???/file name", false },
+            new object[] { "http://www.contoso.com/\u00E4???/file name", false },
+            new object[] { "http:\\host/path/file", false },
+
+            new object[] { "http://www.contoso.com/a/sek http://test.com", false },
+            new object[] { "http://www.contoso.com/\u00E4/sek http://test.com", false },
+
+            new object[] { "http://www.contoso.com/ seka http://test.com", false },
+            new object[] { "http://www.contoso.com/ sek\u00E4 http://test.com", false },
+
+            new object[] { "http://www.contoso.com/ a sek http://test.com", false },
+            new object[] { "http://www.contoso.com/ \u00E4 sek http://test.com", false },
+
+            new object[] { "http://www.contoso.com/ \u00E4/", false },
+            new object[] { "http://www.contoso.com/ path/", false },
+
+            new object[] { "http://www.contoso.com/path", true },
+            new object[] { "http://www.contoso.com/\u00E4/", true },
+
+            new object[] { "http://www.contoso.com/path/#", true },
+            new object[] { "http://www.contoso.com/\u00E4/#", true },
+
+            new object[] { "http://www.contoso.com/path/# ", true },
+            new object[] { "http://www.contoso.com/\u00E4/# ", true },
+
+            new object[] { "http://www.contoso.com/path/ # ", false },
+            new object[] { "http://www.contoso.com/\u00E4/ # ", false },
+
+            new object[] { "http://www.contoso.com/path/ #", false },
+            new object[] { "http://www.contoso.com/\u00E4/ #", false },
+
+            new object[] { "http://www.contoso.com/path ", true },
+            new object[] { "http://www.contoso.com/\u00E4/ ", true },
+
+            new object[] { "http://www.contoso.com/path/\u00E4/path /", false },
+            new object[] { "http://www.contoso.com/path/\u00E4/path / ", false },
+            new object[] { "http://www.contoso.com/path/\u00E4/path/", true },
+            new object[] { "http://www.contoso.com/path/\u00E4 /path/", false },
+            new object[] { "http://www.contoso.com/path/\u00E4 /path/ ", false },
+            new object[] { "http://www.contoso.com/path/\u00E4/path/ \u00E4/", false },
+
+            // Test Query
+            new object[] { "http://www.contoso.com/path?name", true },
+            new object[] { "http://www.contoso.com/path?\u00E4", true },
+
+            new object[] { "http://www.contoso.com/path?name ", true },
+            new object[] { "http://www.contoso.com/path?\u00E4 ", true },
+
+            new object[] { "http://www.contoso.com/path ?name ", false },
+            new object[] { "http://www.contoso.com/path ?\u00E4 ", false },
+
+            new object[] { "http://www.contoso.com/path?par=val?", true },
+            new object[] { "http://www.contoso.com/path?\u00E4=\u00E4?", true },
```

Comment:
@rmkerr is this correct?

Reply:
These look great. Thanks for doing such thorough testing. Just to verify that we're not breaking anything, it would be good to run these test cases without the fix and verify all of the failures are what we expect.